modules/friendlytag.js

```diff
@@ -183,6 +183,248 @@ Twinkle.tag.callback = function friendlytagCallback() {
+				'R from symbol',
+				'R from systematic abbreviation',
+				'R from systematic name',
+				'R from technical name',
+				'R from telephone number',
+				'R from television episode',
+				'R from top-level domain',
+				'R from tour',
+				'R from trade name',
+				'R from trade name',
+				'R from trademark',
+				'R from Twitter username',
+				'R from UK postcode',
+				'R from Unicode character',
+				'R from Unicode code',
+				'R from upcoming film',
+				'R from US military aircraft designator',
+				'R from US postal abbreviation',
+				'R from variant military designation',
+				'R from verb',
+				'R from Wade–Giles romanization',
+				'R from work',
+				'R from writer',
+				'R from year',
+				'R ME from alternative name',
+				'R ME from alternative spelling',
+				'R ME from duplicated article',
+				'R ME from former name',
+				'R ME to list entry',
+				'R ME to section',
+				'R ME with possibilities',
+				'R ME without diacritics',
+				'R plant with possibilities',
+				'R printworthy',
+				'R scouting from merge',
+				'R taxon with possibilities',
+				'R to acronym',
+				'R to article without mention',
+				'R to century',
+				'R to decade',
+				'R to former name',
+				'R to initialism',
+				'R to joint biography',
+				'R to list entry',
+				'R to monotypic taxon',
+				'R to name with title',
+				'R to numerals',
+				'R to plural',
+				'R to related topic',
+				'R to scientific name',
+				'R to sports league',
+				'R to subtopic',
+				'R to Swiss municipality (, Switzerland)',
+				'R to Swiss municipality (canton)',
+				'R to Swiss municipality 2L',
+				'R to systematic name',
+				'R to technical name',
+				'R unprintworthy'
+			];
+			var talkOnly = [
+				'R from remote talk page',
+				'R from comment subpage'
+			];
+			var templateOnly = [
+				'R to stub template',
+				'R to warning template',
+				'R to userbox'
+			];
+			var disambiguationOnly = [
+				'R to disambiguation page'
+			];
+			var userOnly = [
+				'R to user namespace'
+			];
+
+			var deleteTag = function(tagToDelete, tagList) {
+				for (var key1 in tagList) {
+					if (Object.prototype.hasOwnProperty.call(tagList, key1)) {
+						for (var key2 in tagList[key1]) {
+							if (Object.prototype.hasOwnProperty.call(tagList[key1], key2)) {
+								for (var key3 in tagList[key1][key2]) {
+									if (tagList[key1][key2][key3].tag === tagToDelete) {
+										tagList[key1][key2].splice(key3, 1);
+										return;
+									}
+								}
+							}
+						}
+					}
+				}
+			};
+			var deleteTags = function(tagList, redirectList) {
+				tagList.forEach(function(tag) {
+					deleteTag(tag, redirectList);
+				});
+			};
+			// https://stackoverflow.com/a/2548133/3480193
+			var endsWith = function(str, suffix) {
+				return str.indexOf(suffix, str.length - suffix.length) !== -1;
+			};
+
+			var namespace = mw.config.get('wgNamespaceNumber');
+			if (namespace !== 0 && namespace !== 1) { // not mainspace
```

Comment:
These templates are applicable for talk namepace as well? Doesn't look like it: [R from pseudonym](https://en.wikipedia.org/w/index.php?title=Special%3AWhatLinksHere&hidelinks=1&target=Template%3A+R+from+pseudonym&namespace=1), [R from gene symbol](https://en.wikipedia.org/w/index.php?title=Special%3AWhatLinksHere&hidelinks=1&target=Template%3AR+from+gene+symbol&namespace=1)